packages/block-library/src/navigation/index.php

```diff
@@ -248,13 +248,93 @@ function block_core_navigation_render_submenu_icon() {
 	return '<svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" viewBox="0 0 12 12" fill="none" aria-hidden="true" focusable="false"><path d="M1.50002 4L6.00002 8L10.5 4" stroke-width="1.5"></path></svg>';
 }
 
+/**
+ * Get the classic navigation menu to use as a fallback.
+ *
+ * @return object WP_Term The classic navigation.
+ */
+function block_core_navigation_get_classic_menu_fallback() {
+	$classic_nav_menus = wp_get_nav_menus();
+
+	// If menus exist.
+	if ( $classic_nav_menus && ! is_wp_error( $classic_nav_menus ) && count( $classic_nav_menus ) === 1 ) {
+		// If there's only one classic menu then use it.
+		return $classic_nav_menus[0];
+	}
+}
+
+/**
+ * Converts a classic navigation to blocks.
+ *
+ * @param  object WP_Term The classic navigation object to convert.
+ * @return array the normalized parsed blocks.
+ */
+function block_core_navigation_get_classic_menu_fallback_blocks( $classic_nav_menu ) {
+	$menu_items = wp_get_nav_menu_items( $classic_nav_menu->term_id, array( 'update_post_term_cache' => false ) );
+
+	// Set up the $menu_item variables.
+	_wp_menu_item_classes_by_context( $menu_items );
+
+	$sorted_menu_items = array();
+	foreach ( (array) $menu_items as $menu_item ) {
+		$sorted_menu_items[ $menu_item->menu_order ] = $menu_item;
+	}
+
+	unset( $menu_items, $menu_item );
+
+	// END: Code that already exists in wp_nav_menu().
+
+	$menu_items_by_parent_id = array();
+	foreach ( $sorted_menu_items as $menu_item ) {
+		$menu_items_by_parent_id[ $menu_item->menu_item_parent ][] = $menu_item;
+	}
+
+	$inner_blocks = gutenberg_convert_menu_items_to_blocks(
+		isset( $menu_items_by_parent_id[0] )
+			? $menu_items_by_parent_id[0]
+			: array(),
+		$menu_items_by_parent_id
+	);
+
+	return $inner_blocks;
+}
+
+/**
+ * If there's a the classic menu then use it as a fallback.
+ *
+ * @return array the normalized parsed blocks.
+ */
+function block_core_navigation_maybe_use_classic_menu_fallback() {
+	// See if we have a classic menu.
+	$classic_nav_menu = block_core_navigation_get_classic_menu_fallback();
+
+	// If we have a classic menu then convert it to blocks.
+	if ( $classic_nav_menu ) {
+		$classic_nav_menu_blocks            = block_core_navigation_get_classic_menu_fallback_blocks( $classic_nav_menu );
+		$classic_nav_menu_blocks_serialized = serialize_blocks( $classic_nav_menu_blocks );
+
+		// Create a new navigation menu from the classic menu.
+		wp_insert_post(
```

Comment:
Suggestion: this will give us an ID. From that we can [`get_post()`](https://developer.wordpress.org/reference/functions/get_post/). That will _ensure_ we get the exact same post that was created.

Reply:
wp_insert_post can return an error. This function, doesn't always, "just work". Where is the error handling here?

Reply:
For reference https://developer.wordpress.org/reference/functions/wp_insert_post/#return

Reply:
In the future we may not want to get that menu, if the logic for the fallbacks changes. By just calling `block_core_navigation_get_most_recently_published_navigation` again we are letting that function be the authority on what the correct fallback should be, so that if we update the behaviour we won't need to also update the logic here.

Reply:
I've added error handling for the wp_insert_post call.